### PR TITLE
fix github actions deprecation warnings + preflight checks

### PR DESCRIPTION
**Pull request by humbletim**
- update action dependencies to avoid deprecation warnings
- add more error checking and preflight checks around SDK webservice assumptions

### CODE REVIEW

**Comment by humbletim** [comment]
reference points:
- **actions/cache**: https://github.com/actions/cache/issues/953
- **actions/checkout**: https://github.com/actions/checkout/issues/959
- **seanmiddleditch/gha-setup-vsdevenv**: https://github.com/seanmiddleditch/gha-setup-vsdevenv/pull/15

**Comment by humbletim** [comment]
preflight checks TODO:
- [x] improved detection and reporting of webservice errors (including webservice internal error message when available)
  -eg: specifying "bogus" for the SDK release number: image
- [ ] verify config.json ducktypes as a valid Vulkan SDK config before attempting to use

**Comment by edvn0** [comment]
Could we merge this? @humbletim

**Comment by dsvensson** [comment]
Guess a year between reminders is ok @humbletim

**Comment by kalwalt** [comment]
Can you merge this @humbletim ?

**Comment by humbletim** [comment]
Hi.. I am (somewhat) back again and in any event planning to get this PR merged this week. Apologies for the delay. Also if anyone is interested in adopting lead responsibility for the project or helping to co-maintain please let me know.

**Comment by edvn0** [comment]
> Hi.. I am (somewhat) back again and in any event planning to get this PR merged this week. Apologies for the delay. Also if anyone is interested in adopting lead responsibility for the project or helping to co-maintain please let me know.

Don't worry. I could help you out 😃

**Comment by kalwalt** [comment]
> Hi.. I am (somewhat) back again and in any event planning to get this PR merged this week. Apologies for the delay. Also if anyone is interested in adopting lead responsibility for the project or helping to co-maintain please let me know.

Thank you @humbletim ! This is awesome!

File: README.md
Comment (corporateshark) [inline README.md]: `1.3.275.0` is already out.

**Comment by humbletim** [comment]
Hi everyone!

There seems to be a recent issue building SPIRV-Tools on macOS. While unrelated to this PR, I typically run a final "CI build" on different platforms before releasing. Unfortunately, Mac builds are [failing](https://github.com/humbletim/setup-vulkan-sdk/actions/runs/7840660295/job/21395638244) for all but the [the latest SDK version](https://github.com/humbletim/setup-vulkan-sdk/actions/runs/7840660295/job/21395638154), apparent due to `sprintf`-related warnings being promoted into errors by compile flags somewhere. Which for older releases should not have changed... but could have for example on LunarG webservices `config.json` side of the equation.

To understand the scope of the problem, I'd appreciate it if anyone using the SDK on Mac could share:

- Which SDK version are you targeting?
- Have you encountered any recent build failures related to sprintf?

I'm planning to merge this PR to "main" later tonight after one final non-mac-legacy CI test, but am thinking to hold off on stamping an official release tag until I have a clearer picture of the Mac issue. Once I gather more information, I plan to file an issue report and potentially involve relevant teams for a resolution.

Thanks for your input!

**Review by corporateshark — APPROVED** [review]

**Pull request merged**
